packages/framework/fluid-static/api-report/fluid-static.api.md

```diff
@@ -112,7 +107,10 @@ export interface IServiceAudienceEvents<M extends IMember> extends IEvent {
 }
 
 // @public
-export type LoadableObjectClass<T extends IFluidLoadable> = DataObjectClass<T> | SharedObjectClass<T>;
+export type LoadableObjectClass<T extends IFluidLoadable> = FluidStaticEntryPointProvider & LoadableObjectCtor<T>;
+
+// @internal (undocumented)
+export type _LoadableObjectClass<T extends IFluidLoadable> = LoadableObjectCtor<T> & FluidStaticEntryPointProvider<_ProvideChannelFactory & IFluidDataStoreFactory>;
```

Comment:
i think there is also a desire to decouple this from classes, on possible path would be to support branding on the entrypoint so it knows the type, and specifically the interfaces it creates.